ctapipe/reco/HillasReconstructor.py

```diff
@@ -316,11 +346,30 @@ def estimate_core_position(self, hillas_dict, array_pointing):
         core_y: u.Quantity
             estimated y position of impact
 
+        Notes
+        -----
+        The part of the algorithm taking into account divergent pointing
+        mode and the usage of a corrected psi angle is explained in [gasparetto]_
+        section 7.1.4.
+
         """
-        if self.divergent_mode:
-            psi = u.Quantity(list(self.corrected_angle_dict.values()))
-        else:
-            psi = u.Quantity([h.psi for h in hillas_dict.values()])
+
+        # Since psi has been recalculated in the fake CameraFrame
+        # it doesn't need any further corrections because it is now independent
+        # of both pointing and cleaning/parametrization frame.
+        # This angle will be used to visualize the telescope-wise directions of
+        # the shower core the ground.
+        psi_core = self.corrected_angle_dict
```

Comment:
Make this an argument to the method, not a member variable. No reason to store event-wise data on the class.

These methods should all be `@staticmethods` which would also make it easy to unit test those single methods.

Reply:
I guess this solves the related comment above, right?

Reply:
actually this clashes with the hillas planes

also those are event-wise, but are stored because used in different methods

Reply:
Calculate once, give as argument

Reply:
This was a genral comment concerning alls event wise quantities